clang/lib/CIR/CodeGen/CIRGenFunction.cpp

```diff
@@ -1242,6 +1242,18 @@ void CIRGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
       auto address = Address(addr, alignment);
       setAddrOfLocalVar(paramVar, address);
 
+      bool isPromoted = isa<ParmVarDecl>(paramVar) &&
+                        cast<ParmVarDecl>(paramVar)->isKNRPromoted();
+      if (isPromoted) {
+        auto ty = getCIRType(paramVar->getType());
```

Comment:
OG checks that the types are different before emitting any casts. It mentions enums, but I haven't actually been able to trigger a case where that matters. (I commented out that code and `check-clang` still passes.) Maybe just leave a comment about that, so that it's easier to match against OG in the future?

Reply:
Yea, looks like either assert or missing feature use cases.

Reply:
added `assert(!MissingFeatures::constructABIArgDirectExtend());` if you meant this enum